models/webhook/webhook.go

```diff
@@ -150,8 +150,10 @@ func init() {
 // AfterLoad updates the webhook object upon setting a column
 func (w *Webhook) AfterLoad() {
 	w.HookEvent = &webhook_module.HookEvent{}
-	if err := json.Unmarshal([]byte(w.Events), w.HookEvent); err != nil {
-		log.Error("Unmarshal[%d]: %v", w.ID, err)
+	if w.Events != "" {
```

Comment:
In which case is the Event field empty? (outside of the test case)

Wouldn’t it be better to set this field in tests, like in the codebase?

Reply:
> In which case is the Event field empty? (outside of the test case)

I don't see a real case that the Events field would be empty unless some users corrupt their database or there are some legacy/incorrect data.

> Wouldn’t it be better to set this field in tests, like in the codebase?

I think this change is also not bad, at least, it doesn't make anything worse. Meanwhile, when decoding JSON, I think empty string `""` could be safely treated the same as empty object `"{}"`.

----

If the "Events" needs to be fully tested, their values could be set in some following PRs.

Reply:
> it doesn't make anything worse. Meanwhile, when decoding JSON, I think empty string "" could be safely treated the same as empty object "{}".

The current code already treats `""` as an empty object **with an error log**.

I think this log line is valuable and should not be dropped (since such a case means that something went wrong).

Reply:
Gitea has a lot of meaningless error logs, and I think here it doesn't really help anything. At least I don't see a real case how it would help end users, or what end users could do if they see such an error log.

Reply:
> what end users could do if they see such an error log

I would expect the user to investigate to find the root cause for the `Event` field to be empty, since it should never happen.

If the error is silently dropped, the behavior of the webhook will likely be wrong, but the user will have no log line to help debug it.

Reply:
Reverted to old in 3fc6c3e2634ae24ece3faebef14e6715fcce6b48